api/src/main/java/eu/darkbot/api/managers/QuestAPI.java

```diff
@@ -0,0 +1,194 @@
+        int getId();
+
+        /**
+         * @return If the quest is Active
+         */
+        boolean isActive();
+
+        /**
+         * @return The quest category
+         */
+        String getCategory();
+
+        /**
+         * @return The title of the quest in the client's language
+         */
+        String getTitle();
+
+        /**
+         * @return The description of the quest in the client's language
+         */
+        String getDescription();
+
+        /**
+         * @return The number of requirements the quest has to be completed.
+         */
+        int getRequirementsCount();
+
+        /**
+         * @return True if completed
+         */
+        boolean isCompleted();
+
+        /**
+         * @return List of requirements for completion
+         */
+        List<Requirement> getRequirements();
+
+        /**
+         * @return List of rewards to be obtained by completing the quest
+         */
+        List<Reward> getRewards();
+    }
+
+    /**
+     * QuestListItem Object
+     */
+    interface QuestListItem {
+
+        /**
+         * @return The Quest ID
+         */
+        int getId();
+
+        /**
+         * @return Level required to activate it
+         */
+        int getLevelRequired();
+
+        /**
+         * @return True if selected in the interface
+         */
+        boolean isSelected();
+
+        /**
+         * @return True if completed
+         */
+        boolean isCompleted();
+
+        /**
+         * @return Quest Title
+         */
+        String getTitle();
+
+        /**
+         * @return Type of quest
+         */
+        String getType();
+
+        /**
+         * @return True if it can be activated
+         */
+        boolean isActivable();
+    }
+
+    /**
+     * Requirement Object
+     */
+    interface Requirement {
+
+        /**
+         * @return Description of the requirement
+         */
+        String getDescription();
+
+        /**
+         * @return Completed amount of the requirement
+         */
+        double getGoalReached();
+
+        /**
+         * @return Amount needed to complete the requirement
+         */
+        double getGoal();
+
+        /**
+         * @return True if the requirement has been completed
+         */
+        boolean isCompleted();
+
+        /**
+         * @return Type of requirement
+         */
+        String getRequirementType();
+
+        /**
+         * @return Returns the list of requirements that the requirement itself may have
+         */
+        List<Requirement> getRequirements();
+
+        /**
+         * @return Returns whether the requirements can be realised or not.
+         */
+        boolean isEnabled();
+    }
+
+    /**
+     * Reward Object
+     */
+    interface Reward {
+
+        /**
+         * @return Amount to be obtained
+         */
+        int getAmount();
+
+        /**
+         * @return Type to be obtained
+         */
+        String getType();
```

Comment:
Again, is an enum possible?

Reply:
It can be anything from experience, lasers to premium. I don't think it's feasible to make an enum of everything.